### ISSUE TITLE
Support for other Unixes

### ISSUE THREAD

**Issue by bmw** [WoLpH/portalocker#35]
On Unix, `portalocker` uses `flock()` which isn't part of the POSIX standard and is unavailable on Unix systems like [Solaris](https://www.perkin.org.uk/posts/solaris-portability-flock.html). I'd like to use `portalocker` in one of my projects, but we have some users on more obscure Unix systems like Solaris that we don't want to break compatibility with. The alternative to `flock()` is `lockf()` or `fcntl()` which are part of the [POSIX](http://pubs.opengroup.org/onlinepubs/009695399/functions/lockf.html) [standard](http://pubs.opengroup.org/onlinepubs/009695399/functions/fcntl.html) and have the additional benefit of (more likely) [working across NFS](http://0pointer.de/blog/projects/locking.html) better than `flock()`.

Would you be willing to either:

1. Switch to `lockf()` or `fcntl()` over `flock()` on POSIX systems.
2. Fall back to `lockf()` or `fcntl()` if `flock()` is unavailable.

**Comment by WoLpH** [comment]
Switching to `lockf` by default seems like a scary option to me since it might break backwards compatibility. I'm not sure if the behaviour is different, but it is likely to differ on some platforms.

The 2nd option seems like the most sensible one as it will only improve upon the current behaviour. Without access to a solaris system I'm actually not sure what type of exceptions to expect when it's not supported. Do you simply get an `IOError` or does the `flock()` function not exist?

**Comment by bmw** [comment]
Good question! I don't have access to a Solaris system either, but when researching the error you get it looks like there may not be a problem. While `flock()` isn't POSIX compliant and I found articles like the one I linked above about the lack of `flock()` on Solaris, I found a [few](http://infinitesque.net/articles/2010/on%20Python%20flock/) [other](https://lists.gt.net/python/python/99139) [resources](https://bugs.python.org/issue617870) where `fcntl.flock()` appears to work as expected with Python.

I'll probably go ahead and use `portalocker` for [certbot](https://github.com/certbot/certbot/) and if people file bug reports, we can readdress this issue with more information about how the problem manifests. Sorry for the false alarm! Feel free to close this issue.
